embeddings/pipeline/lightning_hps_pipeline.py

```diff
@@ -37,38 +39,21 @@ class _OptimizedLightingPipelineBase(
 
 
 @dataclass
-class OptimizedLightingClassificationPipeline(
+class OptimizedLightingPipeline(
     OptunaPipeline[
-        LightingTextClassificationConfigSpace,
-        LightningClassificationPipelineMetadata,
-        LightningClassificationPipelineMetadata,
+        LightingConfigSpace,
+        LightningPipelineMetadata,
+        LightningPipelineMetadata,
     ],
-    AbstractHuggingFaceOptimizedPipeline[LightingTextClassificationConfigSpace],
-    _OptimizedLightingPipelineBase[LightingTextClassificationConfigSpace],
+    AbstractHuggingFaceOptimizedPipeline[LightingConfigSpace],
+    _OptimizedLightingPipelineBase[LightingConfigSpace],
 ):
     def __post_init__(self) -> None:
-        # Type: ignore is temporal solution due to issue #152 https://github.com/CLARIN-PL/embeddings/issues/152
-        super().__init__(
-            preprocessing_pipeline=HuggingFaceTextClassificationPreprocessingPipeline(
-                dataset_name=self.dataset_name,
-                persist_path=self.tmp_dataset_dir.name,
-                sample_missing_splits=(self.sample_dev_split_fraction, None),
-                ignore_test_subset=True,
-                load_dataset_kwargs=self.load_dataset_kwargs,
-            ),
-            evaluation_pipeline=LightningClassificationPipeline,  # type: ignore
-            pruner=self.pruner_cls(n_warmup_steps=self.n_warmup_steps),
-            sampler=self.sampler_cls(seed=self.seed),
-            n_trials=self.n_trials,
-            dataset_path=self.tmp_dataset_dir.name,
-            metric_name="f1__average_macro",
-            metric_key="f1",
-            config_space=self.config_space,
+        raise NotImplementedError(
+            f"You cannot use generic OptimizedLightingPipeline object. You need to specify pipeline type, e.g. {OptimizedLightingClassificationPipeline.__name__}"
         )
```

Comment:
Agree, abstract class is better solution.

Reply:
https://docs.python.org/3/library/exceptions.html#NotImplementedError

> This exception is derived from RuntimeError. In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method [...].

This is a workaround because mypy does not allow abstract dataclasses. I can change it to abstractmethod and add ignore in this case if you think this is a better option.

Reply:
In my view abstract class It's a better solution, we have used such approach already.

Reply:
ok, no problem